### PR TITLE
add scripts for syncing standalone subprojects code into tree

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes: part of https://github.com/kubeedge/kubeedge/issues/1082

**Special notes for your reviewer**:
Script for syncing beehive code into tree, might be only useful the first time if we only update beehive code from in-tree staging directory afterwards.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hack/sync-subproject-code.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+SPLITER=":"
+if [[ ${USE_HTTPS} ]]; then
+  PROTO_HEAD="https://"
+  SPLITER="/"
+fi
+
+function check-and-add-upstream() {
+  local upstream=$1
+  local upstream_url=$2
+  git remote -v | grep -qE "^${upstream}\b.*${upstream_url}" || rc="$?"
```
Comment (kadisi) [inline hack/sync-subproject-code.sh]: use `git remote get-url ${upstream}` ?

**Comment by sids-b** [comment]
Thanks @kevin-wangzefeng

**Pull request merged**